test/e2e/nstemplatetier_test.go

```diff
@@ -358,27 +361,26 @@ func verifyResourceUpdatesForUserSignups(t *testing.T, hostAwait *HostAwaitility
 			}
 			require.NoError(t, err, "Failing \nUserSignup: %+v \nUserAccount: %+v \nNSTemplateSet: %+v", usersignup, userAccount, nsTemplateSet)
 		}
+
+		// the syncIndex should be different if the tier changed. if only the template refs changed then the sync index is expected to be the same because UserAccounts no longer have references to the NSTemplateSet
 		mur, err := hostAwait.WaitForMasterUserRecord(usersignup.Status.CompliantUsername,
 			UntilMasterUserRecordHasCondition(Provisioned()), // ignore other conditions, such as notification sent, etc.
-			UntilMasterUserRecordHasNotSyncIndex(syncIndex),
+			UntilMasterUserRecordHasSyncIndex(syncIndex, tierNameChanged),
```

Comment:
I figured since we still have the sync logic then it doesn't hurt keeping it just to make sure it's behaving as expected until we remove it but if you prefer to stop checking it entirely then I'm fine doing that too. For these tests we aren't changing the tier name, it's just updates to the templates so it's probably not necessary to check that either?